src/main/scala/rocket/DCache.scala

```diff
@@ -575,7 +574,7 @@ class DCacheModule(outer: DCache) extends HellaCacheModule(outer) {
   tl_out_a.valid := !io.cpu.s2_kill &&
     (s2_valid_uncached_pending ||
       (s2_valid_cached_miss &&
-       !(release_ack_wait && release_ack_dirty) &&
+       !(release_ack_wait && (tl_out_a.bits.address ^ release_ack_addr)(((pgIdxBits + pgLevelBits) min paddrBits) - 1, idxLSB) === 0) &&
```

Comment:
Actually, I think we should follow my original suggestion here and use `s2_req.addr` instead of `tl_out_a.bits.address`.  This clause is already qualified by by `!s2_uncached` (via `s2_valid_cached_miss`), so we know for sure `tl_out_a.bits.address` will be `s2_req.addr`, and we can remove a few levels of logic in the process.